### PR TITLE
Propagate input buffers for passthrough kernels

### PR DESCRIPTION
**Pull request by jbobba**

### CODE REVIEW

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 1]
```diff
@@ -973,6 +974,41 @@ using namespace ngraph::runtime;
...
+
+            if (auto op_annotations = c_op->get_op_annotations())
+            {
+                for (auto oi_pair : op_annotations->get_in_place_oi_pairs())
+                {
+                    if (oi_pair.input == input->get_index() && !oi_pair.destructive)
+                    {
+                        size_t output_index = oi_pair.output;
+                        auto& output_tensor = c_op->get_outputs().at(output_index).get_tensor();
+
```
Comment (Krovatkin) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: Could you please add an `NGRAPH_DEBUG` statement specifying outputs "inplaced"?

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 2]
```diff
@@ -973,6 +974,41 @@ using namespace ngraph::runtime;
...
+                        m_variable_name_map[output_tensor.get_name()] = input_name;
+                        it = &c_op->get_outputs().at(output_index);
+                        propagate_further = true;
```
Comment (Krovatkin) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: if we have more than one non-destructive users we will only propagate parameter's storage through one of them? I think we might want consider using a work stack here.

File: src/ngraph/runtime/cpu/cpu_external_function.cpp (hunk 1 above)
Comment (jbobba) [reply src/ngraph/runtime/cpu/cpu_external_function.cpp]: Will sneak it in one of the other PRs.

File: src/ngraph/runtime/cpu/cpu_external_function.cpp (hunk 2 above)
Comment (jbobba) [reply src/ngraph/runtime/cpu/cpu_external_function.cpp]: good point. will address

**Pull request merged**